### ISSUE TITLE
[zos-files] `upload_file_to_dsn` does not convert Windows line endings

### ISSUE THREAD

**Issue by t1m0thyj** [zowe/zowe-client-python-sdk#104]
When uploading a file with Windows line endings, the `\r\n` line endings turn into double line endings.

The expected behavior is that (unless the file is uploaded in binary mode) the line endings would be converted to Unix (`\n`).

**Comment by t1m0thyj** [comment]
The following snippet can be used to test uploading/downloading a file for this issue (and #179).
```python
from decouple import config
from zowe.zos_files_for_zowe_sdk import Files

profile = {
    "host": "<zosmfHost>",
    "port": 443,
    "rejectUnauthorized": True,
    "user": config("ZOWE_OPT_USER"),
    "password": config("ZOWE_OPT_PASSWORD")
}
dsname = "<testDsname>"
files_api = Files(profile)
files_api.upload_file_to_dsn("test-upload.txt", dsname)
files_api.download_dsn(dsname, "test-download.txt")
```

**Comment by t1m0thyj** [comment]
I think in order to fix this, we'll need to add a "binary" option to the `upload_file_to_dsn` method.

When this option is true, we can open the input file in `rb` mode and upload it without any processing.

When this option is false, we can open the file in `r` mode which will normalize new lines before uploading - according to [the docs](https://docs.python.org/3.10/library/functions.html#open) Python's default behavior is to replace `\r\n` with `\n`.

**Comment by adam-wolfe** [comment]
Fix implemented in #276
